src/vulkan/pipeline.cc

```diff
@@ -268,18 +265,42 @@ Result Pipeline::AddDescriptor(const BufferCommand* cmd) {
   descriptor_set_info_[desc_set].empty = false;
 
   auto& descriptors = descriptor_set_info_[desc_set].descriptors;
-  Descriptor* desc = nullptr;
   for (auto& descriptor : descriptors) {
-    if (descriptor->GetBinding() == cmd->GetBinding())
-      desc = descriptor.get();
+    if (descriptor->GetBinding() == binding)
+      *desc = descriptor.get();
   }
 
+  return {};
+}
+
+Result Pipeline::AddDescriptor(const BufferCommand* cmd) {
```

Comment:
Can we rename this to `AddBufferDescriptor` and use `AddSamplerDescriptor` below?

Reply:
Done